packages/backend/src/models/SavedChartModel.ts

```diff
@@ -963,6 +964,115 @@ export class SavedChartModel {
         }
     }
 
+    async findChartsForValidation(projectUuid: string): Promise<
+        Array<{
+            uuid: string;
+            name: string;
+            tableName: string;
+            filters: Filters;
+            dimensions: string[];
+            metrics: string[];
+            tableCalculations: string[];
+            customMetrics: string[];
+            customMetricsBaseDimensions: string[];
+            customBinDimensions: string[];
+            customSqlDimensions: string[];
+            sorts: string[];
+        }>
+    > {
+        return this.database(SavedChartsTableName)
+            .select({
+                uuid: 'saved_queries.saved_query_uuid',
+                name: 'saved_queries.name',
+                tableName: 'saved_queries_versions.explore_name',
+                filters: 'saved_queries_versions.filters',
+                dimensions: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT svf.name) FILTER (WHERE svf.field_type = 'dimension'), '{}')",
+                ),
+                metrics: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT svf.name) FILTER (WHERE svf.field_type = 'metric'), '{}')",
+                ),
+                tableCalculations: this.database.raw(
+                    "COALESCE(ARRAY_AGG(DISTINCT sqvtc.name) FILTER (WHERE sqvtc.name IS NOT NULL), '{}')",
```

Comment:
do you know if all these `where`fields, (eg: `sqvtc.name` ) are indexes ? If not we can make them, or leave a comment to do it in case we notice some performace issues.

Reply:
Good question. I assume this filter is only applied to the values after the main where clause. So it will filter the table calculations for the latest chart version.

Example:

Org table

| id  | name |
| ------------- | ------------- |
| 1 | A  |
| 2  | B  |

User table

| id  | name | org_id |
| ------------- | ------------- | ------------- |
| 1 | Jose  | 1 |
| 2  | Javier  | 1 |
| 3  | Katie  | 2 |
| 4  | Jake  | null |
| 5  | null  | 1 |

SQL
```sql
SELECT
  orgs.name, 
  COALESCE(ARRAY_AGG(DISTINCT users.name) FILTER (WHERE users.name IS NOT NULL), '{}') as members
FROM orgs
  LEFT JOIN users ON users.org_id = orgs.id
WHERE 
   org.id = 1
```

The `FILTER (WHERE users.name IS NOT NULL)` would only filter the rows that matches the org id filter:

| id  | name | org_id |
| ------------- | ------------- | ------------- |
| 1 | Jose  | 1 |
| 2  | Javier  | 1 |
| 5  | null  | 1 |

Final results:

| name  | members |
| ------------- | ------------- |
| A | { Jose, Javier } | 


@TuringLovesDeathMetal  @jpetey75 Can one of you confirm this?

Reply:
Ya, `WHERE` is executed _**after**_ the join

if you want it to execute _**before**_ the join, then you'd need to add it as logic in the `join` clause. 

E.g. these would result in the same table (but I feel like there are probably examples where this wouldn't result int he same table...I haven't thought too hard about this!)

```sql
SELECT
  orgs.name, 
  COALESCE(ARRAY_AGG(DISTINCT users.name) FILTER (WHERE users.name IS NOT NULL), '{}') as members
FROM orgs
  LEFT JOIN users ON users.org_id = orgs.id
WHERE 
   org.id = 1
   ```
   
   ```sql
   SELECT
  orgs.name, 
  COALESCE(ARRAY_AGG(DISTINCT users.name) FILTER (WHERE users.name IS NOT NULL), '{}') as members
FROM orgs
  LEFT JOIN users ON users.org_id = orgs.id AND org.id = 1
   ```